spring-xd-dirt/src/main/java/org/springframework/xd/dirt/job/dsl/Tokenizer.java

```diff
@@ -0,0 +1,388 @@
+
+	private String expressionString; // The string to be tokenized
+
+	private char[] toProcess; // The expressionString as a char array
+
+	private int max; // Length of input data
+
+	private int pos; // Current lexing position in the input data
+
+	private List<Token> tokens = new ArrayList<Token>(); // Output stream of tokens
+
+	public Tokenizer(String inputdata) {
+		this.expressionString = inputdata;
+		this.toProcess = (inputdata + "\0").toCharArray();
+		this.max = toProcess.length;
+		this.pos = 0;
+		process();
+	}
+
+	private void process() {
+		boolean justProcessedEquals = false;
+		while (pos < max) {
+			char ch = toProcess[pos];
+
+			if (justProcessedEquals) {
+				if (!isWhitespace(ch) && ch != 0) {
+					// following an '=' we commence a variant of regular tokenization,
+					// here we consume everything up to the next special char.
+					// This allows SpEL expressions to be used without quoting in many
+					// situations.
+					lexArgValueIdentifier();
+				}
+				justProcessedEquals = false;
+				continue;
+			}
+
+			if (isAlphabetic(ch) || isDigit(ch) || ch == '_' || ch == '.') {
+				lexIdentifier();
+			}
+			else {
+				switch (ch) {
+					case '&':
+						pushCharToken(TokenKind.AMPERSAND);
+						break;
+					case '<':
+						pushCharToken(TokenKind.SPLIT_OPEN);
+						break;
+					case '>':
+						pushCharToken(TokenKind.SPLIT_CLOSE);
+						break;
+					case '|':
+						if (isTwoCharToken(TokenKind.DOUBLE_PIPE)) {
+							pushPairToken(TokenKind.DOUBLE_PIPE);
+						}
+						else {
+							pushPairToken(TokenKind.PIPE);
+						}
+						break;
+					case '-':
+						if (!isTwoCharToken(TokenKind.DOUBLE_MINUS)) {
+							throw new JobSpecificationException(
+									expressionString, pos,
+									JobDSLMessage.MISSING_CHARACTER, "-");
+						}
+						pushPairToken(TokenKind.DOUBLE_MINUS);
+						break;
+					case '=':
+						justProcessedEquals = true;
+						pushCharToken(TokenKind.EQUALS);
+						break;
+					case ' ':
+					case '\t':
+					case '\r':
+						// drift over white space
+						pos++;
+						break;
+					case '\n':
+						pos++;
+						//						pushCharToken(TokenKind.NEWLINE);
+						break;
+					//					case '.':
+					//						pushCharToken(TokenKind.DOT);
+					//						break;
+					//					case ':':
+					//						pushCharToken(TokenKind.COLON);
+					//						break;
+					case '(':
+						pushCharToken(TokenKind.OPEN_PAREN);
+						break;
+					case ')':
+						pushCharToken(TokenKind.CLOSE_PAREN);
+						break;
+					//					case ';':
+					//						pushCharToken(TokenKind.SEMICOLON);
+					//						break;
+					case '\'':
+						lexQuotedStringLiteral();
+						break;
+					case '"':
+						lexDoubleQuotedStringLiteral();
+						break;
+					//					case '@':
+					//						pushCharToken(TokenKind.REFERENCE);
+					//						break;
+					case 0:
+						// hit sentinel at end of char data
+						pos++; // will take us to the end
+						break;
+					case '\\':
+						throw new StreamDefinitionException(
```

Comment:
I'm probably being doing nitpicky here... but would we want to have a JobDefinitionException.  Feel free to argue with me. ;-)